### DISCUSSION TITLE
Separating types from the core

### DISCUSSION THREAD

**Discussion by olivermrbl** [medusajs/medusa#2837] [Feature Requests]
### What
Leverage our OAS to generate types and ship them separately from the core

### Motivation
**Main motivation:** Manually maintaining API clients is time-consuming and error-prone. Time that could be spent on more valuable efforts

**Secondary motivation:** Reduce direct imports from Medusa core package in clients and apps. These direct imports sometimes have side effects, likes pulling unnecessary dependencies into clients and apps.

**Comment by ltran1612** [comment]
I'm new to medusa so I'm sorry if it's a dumb question, but which package should I install instead of "@medusajs/medusa" just for the types?
Thanks!

**Comment by pevey** [comment]
This would be awesome.  This is the primary thing that scared me away from using medusa-js in my storefront app.  The number of dependencies it would have introduced was overwhelming to me.

**Comment by adamlamaa** [comment]
I will echo everybody above me, in that I am also looking for what package to install to avoid installing the full `@medusajs/medusa` package in my front end where I just need type definitions.

**Reply by olivermrbl** [reply]
@adamlamaa, please check out `@medusa/client-types` and let me know if it suffices :)

**Reply by adamlamaa** [reply]
@olivermrbl Thank you.

I was able to replace `medusajs/medusa` with `@medusajs/client-types`

Run: 'yarn add --dev @medusajs/client-types' to install.

**Reply by edardev** [reply]
![image](https://github.com/medusajs/medusa/assets/21114220/20e943a7-fa5d-4b3d-a8c0-4e040bea3a30)

![image](https://github.com/medusajs/medusa/assets/21114220/974c7691-0cad-40f8-a592-d0bb2e039ec9)

Why the nextjs-starter-medusa still use the @medusajs/medusa-js? I am confused in this one..

**Reply by davidparys** [reply]
How did you manage to do that? 

```
Cannot find module '@medusa/client-types' or its corresponding type declarations.ts(2307)
```

I absolutely have no idea how to import these anywhere.

**Reply by imash96** [reply]
if you are using npm "npm i -D @medusajs/client-types"

There is no use of @medusajs/client-types becuase it is not compatible with "medusajs/medusa-js". All the types are mismatch and if you are using "@medusajs/medusa-js" in your storefront you have have to use @medusajs/medusa for types. @olivermrbl correct me if i am wrong.

**Reply by davidparys** [reply]
Lovely! I was confused because `medusajs/medusa-js` has `@medusajs/medusa` and I somehow could not import it. Installing `@medusajs/medusa` solved it (though feels a bit unnecessary but whatever!)

Thanks!
